Common/src/CConfig.cpp

```diff
@@ -355,6 +355,17 @@ void CConfig::addEnumOption(const string name, unsigned short & option_field, co
   return;
 }
 
+// enum types work differently than all of the others because there are a small number of valid
+// string entries for the type. One must also provide a list of all the valid strings of that type.
+template <class Tenum>
+void CConfig::addEnumClassOption(const string name, Tenum & option_field, const map<string, Tenum> & enum_map, Tenum default_value) {
```

Comment:
I just swapped the `Tenum & option_field` from `unsigned short & option_field` and had to pull that through to `COptionEnumClass` ... it is code duplication but there is a more elegant solution without redoing more then feel free to revert and push, I am interested. 
All the `Kind_stuff` vars are unsigned shorts .. if they were their respective enum type the `addEnumClassOption` could handle those as well within the same implementation.

Reply:
ups ... didn't see you already commented when I wrote this